### PR TITLE
feat(query): Added Secret Without Expiration Date query for ARM

### PR DESCRIPTION
**Pull request by joaoReigota1**
Signed-off-by: João Reigota <joao.reigota@checkmarx.com>

**Proposed Changes**
- Added Secret Without Expiration Date query for ARM

All Secrets must have an expiration date defined

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: assets/queries/azureResourceManager/secret_without_expiration_date/query.rego [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package Cx
+
```
Comment (rogeriopeixotocx) [inline assets/queries/azureResourceManager/secret_without_expiration_date/query.rego:2]:
```suggestion
import data.generic.common as commonLib
```

File: assets/queries/azureResourceManager/secret_without_expiration_date/query.rego [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+CxPolicy[result] {
+	resource := input.document[i].resources[_]
+	resource.type == "Microsoft.KeyVault/vaults/secrets"
+	object.get(resource.properties, "attributes", "undefined") == "undefined"
```
Comment (rogeriopeixotocx) [inline assets/queries/azureResourceManager/secret_without_expiration_date/query.rego:6]:
```suggestion
	not commonLib.valid_key(resource.properties, "attributes")
```

File: assets/queries/azureResourceManager/secret_without_expiration_date/query.rego [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		"issueType": "MissingAttribute",
+		"keyExpectedValue": "resource with type 'Microsoft.Security/securityContacts' has 'attributes.exp' property id defined",
+		"keyActualValue": "resource with type 'Microsoft.Security/securityContacts' doesn't have 'attributes' property defined",
+	}
+}
+
+CxPolicy[result] {
+	resource := input.document[i].resources[_]
+	resource.type == "Microsoft.KeyVault/vaults/secrets"
+	object.get(resource.properties.attributes, "exp", "undefined") == "undefined"
```
Comment (rogeriopeixotocx) [inline assets/queries/azureResourceManager/secret_without_expiration_date/query.rego:20]:
```suggestion
    not commonLib.valid_key(resource.properties, "exp")
```

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]

File: assets/queries/azureResourceManager/secret_without_expiration_date/metadata.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
+  "id": "cff9c3f7-e8f0-455f-9fb4-5f72326da96e",
+  "queryName": "Secret Without Expiration Date",
+  "severity": "HIGH",
+  "category": "Best Practices",
+  "descriptionText": "All Secrets must have an expiration date defined",
+  "descriptionUrl": "https://docs.microsoft.com/en-us/azure/templates/microsoft.keyvault/vaults/secrets?tabs=json#SecretAttributes",
+  "platform": "AzureResourceManager",
+  "cloudProvider": "common",
```
Comment (rafaela-soares) [inline assets/queries/azureResourceManager/secret_without_expiration_date/metadata.json:9]: `"cloudProvider": "common"`

**Review by rafaela-soares — CHANGES_REQUESTED** [review]

**Review by rogeriopeixotocx — APPROVED** [review]
LGTM

**Review by rafaela-soares — APPROVED** [review]
LGTM

**Pull request merged**